src/core/src/op/select.cpp

```diff
@@ -47,111 +76,73 @@ void op::v1::Select::validate_and_infer_types() {
-              const HostTensorVector& input_values,
-              const op::AutoBroadcastSpec& autob) {
-    using T = typename element_type_traits<ET>::value_type;
-
-    const auto& in_cond = input_values[0];
-    const auto& in_then = input_values[1];
-    const auto& in_else = input_values[2];
-
-    const auto& out = output_values[0];
-
-    ov::reference::select<T>(in_cond->get_data_ptr<char>(),
-                             in_then->get_data_ptr<T>(),
-                             in_else->get_data_ptr<T>(),
-                             out->get_data_ptr<T>(),
-                             in_cond->get_shape(),
-                             in_then->get_shape(),
-                             in_else->get_shape(),
-                             autob);
-    return true;
-}
-
-bool evaluate_select(const HostTensorVector& output_values,
-                     const HostTensorVector& input_values,
-                     const op::AutoBroadcastSpec& autob,
-                     const element::Type_t& et) {
-    bool rc = false;
-
-    switch (et) {
-        OPENVINO_TYPE_CASE(evaluate_select, i8, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, i16, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, i32, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, i64, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, u8, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, u16, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, u32, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, u64, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, bf16, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, f16, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, f32, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, f64, output_values, input_values, autob);
-        OPENVINO_TYPE_CASE(evaluate_select, boolean, output_values, input_values, autob);
-    default:
-        rc = false;
-        break;
-    }
-
-    return rc;
-}
-}  // namespace
-}  // namespace detail
-
-bool op::v1::Select::evaluate(const HostTensorVector& output_values, const HostTensorVector& input_values) const {
+bool Select::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v1_Select_evaluate);
-    OPENVINO_SUPPRESS_DEPRECATED_START
-    OPENVINO_ASSERT(validate_host_tensor_vector(input_values, 3));
-    OPENVINO_ASSERT(validate_host_tensor_vector(output_values, 1));
-    OPENVINO_SUPPRESS_DEPRECATED_END
-    const auto autob = get_auto_broadcast();
-    return detail::evaluate_select(output_values, input_values, autob, output_values[0]->get_element_type());
+    OPENVINO_ASSERT(inputs.size() == 3);
```

Comment:
is this check required if `shape_infer()` used?

Reply:
Right, it's not needed - removed